src/DelayId.cc

```diff
@@ -40,7 +40,7 @@ DelayId::~DelayId ()
 void
 DelayId::compositePosition(DelayIdComposite::Pointer newPosition)
 {
-    compositeId = newPosition;
+    compositeId = std::move(newPosition);
```

Comment:
Similar to earlier PRs, we do not want to sprinkle code with std::move calls. In this particular case, I am not sure whether any code changes are required (because I am not sure whether the compiler can figure it all out without our help), but I would also investigate whether we should make newPosition a constant reference (and inline this method?).

Reply:
Moved to const-ref. I'm not sure about inlining, I'll skip that for now